app/controllers/api/groups_controller.rb

```diff
@@ -305,6 +305,54 @@ def remove_tag
       render 'shared/http_status', locals: { code: '404', message: I18n.t('tags.not_found') }, status: :not_found
     end
 
+    def extension
+      grouping = Grouping.find_by(group_id: params[:id], assignment: params[:assignment_id])
+      case request.method
+      when 'DELETE'
+        if grouping.extension.present?
+          grouping.extension.destroy!
+          # Successfully deleted the extension; render success
+          render 'shared/http_status', locals: { code: '200', message:
+            HttpStatusHelper::ERROR_CODE['message']['200'] }, status: :ok
+        else
+          # cannot delete a non existent extension; render failure
+          render 'shared/http_status', locals: { code: '422', message:
+            HttpStatusHelper::ERROR_CODE['message']['422'] }, status: :unprocessable_entity
+          nil
+        end
+      when 'POST'
+        extension_values = extension_params
+        extension_values[:time_delta] = time_delta_params if extension_values[:time_delta].present?
+        if grouping.extension.nil?
+          grouping.create_extension!(extension_values)
+          # Successfully created the extension record; render success
+          render 'shared/http_status', locals: { code: '201', message:
+            HttpStatusHelper::ERROR_CODE['message']['201'] }, status: :created
+        else
+          # cannot create extension as it already exists; render failure
+          render 'shared/http_status', locals: { code: '422', message:
+            HttpStatusHelper::ERROR_CODE['message']['422'] }, status: :unprocessable_entity
+          nil
```

Comment:
This is a bit strange, I'm not sure why you've added these `nil`s?

Reply:
Yes, you are correct, not sure how it got there. Maybe my copy and paste accidentally had that value. Thanks for catching this.